data/actions/scripts/other/potions.lua

```diff
@@ -1,158 +1,149 @@
+function onUse(player, potion, fromPosition, target, toPosition, isHotkey)
 	if target == nil or not target:isPlayer() then
 		return true
 	end
-
+	
 	if player:getCondition(CONDITION_EXHAUST_HEAL) then
 		player:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_YOUAREEXHAUSTED))
 		return true
 	end
 
-	local itemId = item:getId()
-	if itemId == antidotePot then
-		if not antidote:execute(target, numberToVariant(target:getId())) then
-			return false
-		end
-
+	local stats = potions[potion:getId()]
+	if stats then
+		if (stats.vocations or stats.minLevel) and not player:getGroup():getAccess() then
+			if not stats.minLevel then
+				stats.minLevel = 1
+			end
+			if not stats.vocations then
+				stats.vocations = {}
+			end
+			
+			if not inArray(states.vocations, player:getVocation():getId()) or player:getLevel() < stats.minLevel then
+				local str, vocations = "", {}
+				-- We only show unpromoted vocations in description.
+				for k, v in ipairs(stats.vocations) do
+					if Vocation(v):getDemotion() then
+						table.insert(vocations, v)
+					end
+				end
+				if #vocations > 0 then
+					for i = 1, #vocations do
+						local voc = Vocation(vocations[i])
+						if not voc:getDemotion() then -- Elder druid won't be showed;
+							if i == #vocations then
+								str = str .. (#vocations ~= 1 and " and " or "") .. voc:getName():lower() .. "s"
+							else
+								str = str .. (i ~= 1 and ", " or "") .. voc:getName():lower() .. "s"
+							end
+						end
+					end
+				end
+				vocations = nil
+				if str == "" then
+					str = "players"
+				end
+				player:say("This potion can be consumed by ".. str .. " of level " .. stats.minLevel .. " or higher.", TALKTYPE_MONSTER_SAY)
+				return false
+			end
+		end
+		if stats.health then
+			if not doTargetCombatHealth(0, target, COMBAT_HEALING, stats.health.min, stats.health.max, CONST_ME_MAGIC_BLUE) then
+				return false
+			end
+		end
+		if stats.mana then
+			if not doTargetCombatMana(0, target, stats.mana.min, stats.mana.max, CONST_ME_MAGIC_BLUE) then
+				return false
+			end
+		end
+		
+		if stats.combat then
+			if not stats.combat:execute(target, numberToVariant(target:getId())) then
+				return false
+			end
+		end
+		
 		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(emptyPot, 1)
-	elseif itemId == smallHealthPot then
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 60, 90, CONST_ME_MAGIC_BLUE) then
-			return false
+		if potions.words then
+			target:say(potions.words, TALKTYPE_MONSTER_SAY)
 		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(emptyPot, 1)
-	elseif itemId == healthPot then
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 125, 175, CONST_ME_MAGIC_BLUE) then
-			return false
+		if potions.removePots then
+			item:remove(1)
```

Comment:
How you pretend to use `item:` if you removed from `onUse` args? it will be nil.